### PR TITLE
[Magiclysm] Change the Utility Molding spell to allow selecting the starting tool

### PR DESCRIPTION
**Pull request by Standing-Storm**
#### Summary
Mods "Change the Utility Molding spell to allow selecting the starting tool"

#### Purpose of change

run_eoc_selector exists

#### Describe the solution

Change the Utility Molding spell so that rather than always summoning a hammer and requiring the player to activate to the proper tool, allow the player to choose which tool is summoned when the spell is first cast. Other tools are removed, so the player can only have one tool at a time. The previous behavior of allowing the tool to transform is retained (I did not replace the transform function on the items because EoC use_actions lack fields like "need_wielding" )

#### Describe alternatives you've considered

Doing nothing

#### Testing
image

Everything works.

### CODE REVIEW

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 1]
```diff
@@ -503,6 +503,160 @@
...
+      { "u_cast_spell": { "id": "eshaper_spawn_shovel" } }
+    ]
+  },
+  {
+    "id": "eshaper_spawn_hammer",
+    "type": "SPELL",
+    "name": "Utility Molding (Hammer)",
+    "description": "Spawn the Utility molding hammer.  It's a bug if you have this spell directly.",
+    "valid_targets": [ "self" ],
+    "flags": [ "SOMATIC", "LOUD" ],
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:591]:
```suggestion
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 2]
```diff
@@ -503,6 +503,160 @@
...
+    "min_damage": 1,
+    "max_damage": 1,
+    "effect": "spawn_item",
+    "effect_str": "eshaper_hammer",
+    "shape": "blast",
+    "spell_class": "EARTHSHAPER",
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:597]:
```suggestion
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 3]
```diff
@@ -503,6 +503,160 @@
...
+    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
+  },
+  {
+    "id": "eshaper_spawn_pickaxe",
+    "type": "SPELL",
+    "name": "Utility Molding (Pickaxe)",
+    "description": "Spawn the Utility molding pickaxe.  It's a bug if you have this spell directly.",
+    "valid_targets": [ "self" ],
+    "flags": [ "SOMATIC", "LOUD" ],
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:606]:
```suggestion
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 4]
```diff
@@ -503,6 +503,160 @@
...
+    "min_damage": 1,
+    "max_damage": 1,
+    "effect": "spawn_item",
+    "effect_str": "eshaper_pickaxe",
+    "shape": "blast",
+    "spell_class": "EARTHSHAPER",
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:612]:
```suggestion
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 5]
```diff
@@ -503,6 +503,160 @@
...
+    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
+  },
+  {
+    "id": "eshaper_spawn_knife",
+    "type": "SPELL",
+    "name": "Utility Molding (Knife)",
+    "description": "Spawn the Utility molding knife.  It's a bug if you have this spell directly.",
+    "valid_targets": [ "self" ],
+    "flags": [ "SOMATIC", "LOUD" ],
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:636]:
```suggestion
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 6]
```diff
@@ -503,6 +503,160 @@
...
+    "min_damage": 1,
+    "max_damage": 1,
+    "effect": "spawn_item",
+    "effect_str": "eshaper_knife",
+    "shape": "blast",
+    "spell_class": "EARTHSHAPER",
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:642]:
```suggestion
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 7]
```diff
@@ -503,6 +503,160 @@
...
+    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
+  },
+  {
+    "id": "eshaper_spawn_shovel",
+    "type": "SPELL",
+    "name": "Utility Molding (Shovel)",
+    "description": "Spawn the Utility molding shovel.  It's a bug if you have this spell directly.",
+    "valid_targets": [ "self" ],
+    "flags": [ "SOMATIC", "LOUD" ],
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:651]:
```suggestion
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 8]
```diff
@@ -503,6 +503,160 @@
...
+    "min_damage": 1,
+    "max_damage": 1,
+    "effect": "spawn_item",
+    "effect_str": "eshaper_shovel",
+    "shape": "blast",
+    "spell_class": "EARTHSHAPER",
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:657]:
```suggestion
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 9]
```diff
@@ -503,6 +503,160 @@
...
+    "description": "Spawn the Utility molding hammer.  It's a bug if you have this spell directly.",
+    "valid_targets": [ "self" ],
+    "flags": [ "SOMATIC", "LOUD" ],
+    "min_damage": 1,
+    "max_damage": 1,
+    "effect": "spawn_item",
+    "effect_str": "eshaper_hammer",
+    "shape": "blast",
+    "spell_class": "EARTHSHAPER",
+    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:598]:
```suggestion
    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] },
    "max_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 10]
```diff
@@ -503,6 +503,160 @@
...
+    "description": "Spawn the Utility molding pickaxe.  It's a bug if you have this spell directly.",
+    "valid_targets": [ "self" ],
+    "flags": [ "SOMATIC", "LOUD" ],
+    "min_damage": 1,
+    "max_damage": 1,
+    "effect": "spawn_item",
+    "effect_str": "eshaper_pickaxe",
+    "shape": "blast",
+    "spell_class": "EARTHSHAPER",
+    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:613]:
```suggestion
    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] },
    "max_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 11]
```diff
@@ -503,6 +503,160 @@
...
+  },
+  {
+    "id": "eshaper_spawn_axe",
+    "type": "SPELL",
+    "name": "Utility Molding (Axe)",
+    "description": "Spawn the Utility molding axe.  It's a bug if you have this spell directly.",
+    "valid_targets": [ "self" ],
+    "flags": [ "SOMATIC", "LOUD" ],
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:621]:
```suggestion
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 12]
```diff
@@ -503,6 +503,160 @@
...
+    "min_damage": 1,
+    "max_damage": 1,
+    "effect": "spawn_item",
+    "effect_str": "eshaper_ax",
+    "shape": "blast",
+    "spell_class": "EARTHSHAPER",
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:627]:
```suggestion
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 13]
```diff
@@ -503,6 +503,160 @@
...
+    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:628]:
```suggestion
    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] },
    "max_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 14]
```diff
@@ -503,6 +503,160 @@
...
+    "description": "Spawn the Utility molding knife.  It's a bug if you have this spell directly.",
+    "valid_targets": [ "self" ],
+    "flags": [ "SOMATIC", "LOUD" ],
+    "min_damage": 1,
+    "max_damage": 1,
+    "effect": "spawn_item",
+    "effect_str": "eshaper_knife",
+    "shape": "blast",
+    "spell_class": "EARTHSHAPER",
+    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:643]:
```suggestion
    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] },
    "max_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 15]
```diff
@@ -503,6 +503,160 @@
...
+    "description": "Spawn the Utility molding shovel.  It's a bug if you have this spell directly.",
+    "valid_targets": [ "self" ],
+    "flags": [ "SOMATIC", "LOUD" ],
+    "min_damage": 1,
+    "max_damage": 1,
+    "effect": "spawn_item",
+    "effect_str": "eshaper_shovel",
+    "shape": "blast",
+    "spell_class": "EARTHSHAPER",
+    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:658]:
```suggestion
    "min_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] },
    "max_duration": { "math": [ "( (u_val('spell_level', 'spell: eshaper_spawn_tools') * 576000) + 576000)" ] }
```

**Review by GuardianDll — APPROVED** [review]

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 16]
```diff
@@ -503,6 +503,160 @@
...
+          "Transform the living earth into a shovel."
+        ]
+      }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_EARTHSHAPER_UTILITY_MOLDING_HAMMER",
+    "effect": [
+      { "u_remove_item_with": "eshaper_pickaxe" },
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:534]: Maybe not remove molding tools if you cast a spell? what if someone need axe and knife simultaneously?

Comment (Standing-Storm) [reply data/mods/Magiclysm/Spells/earthshaper.json:534]: Was it previously possible to have multiple items?  I assumed it wasn't--if it was then you're right, there's no reason to remove existing ones

Comment (Standing-Storm) [reply data/mods/Magiclysm/Spells/earthshaper.json:534]: Looks like the original does allow for multiple tools.  I'll take out the remove_items

**Comment by GuardianDll** [comment]
I also recommend to change flags for spells from `[ "SOMATIC", "LOUD" ]` to `[ "SILENT", "NO_EXPLOSION_SFX" ]`, remove `spell_class`, and add `max_duration` so it actually work, (also, isn't lack of `max_duration` should cause inability to spawn item?)
it was some of the stuff i suggested, but updating the branch marked it as resolved

**Comment by Standing-Storm** [comment]
For the record the items spawned just fine with only a min_duration set during my testing--as far as I've been able to determine, if you use a math statement to automatically scale the duration, you only need a min_duration

**Pull request merged**